### PR TITLE
Feature/drag and drop DAO-637 DAO-649 

### PR DESCRIPTION
**Pull request by sepehr2github**
- react dropzone package added
- input avatar file added
- validation logic added

### CODE REVIEW

**Review by RakeshUP — APPROVED** [review]

File: packages/ui-components/src/components/input/fileAvatarInput.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+   * limit maximum file size of the image (in bytes)
+   */
+  maxFileSize: number;
+};
+
+export const FileAvatarInput: React.FC<FileAvatarInputProps> = ({
+  onChange,
+  maxDimension = 2400,
+  minDimension = 256,
+  maxFileSize = 5000000,
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/input/fileAvatarInput.tsx]:
```suggestion
  maxFileSize = 3000000,
```

Acceptance criteria changed :)

File: packages/ui-components/src/components/input/fileAvatarInput.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    maxSize: maxFileSize,
+    accept: 'image/jpg, image/jpeg, image/png, image/gif, image/svg',
+  });
+  return (
+    <>
+      {loading ? (
+        <LoadingContainer>
+          <Spinner size="small" />
+        </LoadingContainer>
+      ) : (
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/input/fileAvatarInput.tsx]: Small syntax suggestion: You can directly chain a new tertiary operator: `) : preview ? (`. Also, you don't need the extra `<>...</>` block

File: packages/ui-components/src/components/input/fileAvatarInput.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+})``;
+
+const Preview = styled.img.attrs({
+  className: 'rounded-xl bg-ui-0 h-8 w-8',
+})``;
+
+const StyledButton = styled(ButtonIcon).attrs({
+  className: 'absolute',
+})`
+  top: -10px;
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/input/fileAvatarInput.tsx]:
```suggestion
  className: 'absolute -top-2 -right-1.75 ',
```
And then you can delete `top` and `left` on line 139 and 140

File: packages/ui-components/src/components/input/fileAvatarInput.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  maxDimension = 2400,
+  minDimension = 256,
+  maxFileSize = 5000000,
+  onError,
+}) => {
+  const [preview, setPreview] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const onDrop = useCallback(
+    (acceptedFiles: Array<File>, onDropRejected) => {
```
Comment (WissenIstNacht) [inline packages/ui-components/src/components/input/fileAvatarInput.tsx]: (Why) Does this have to be an array?

**Review by WissenIstNacht — CHANGES_REQUESTED** [review]
Hey Sepehr, nice PR, thanks! :)

I left a few minor suggestion. Also, I suggested on Figma that this component be renamed `InputImage`. I hope this will be sorted by the time you get back from vacation.

File: packages/ui-components/src/components/input/fileAvatarInput.tsx (hunk 4 above)
Comment (Fabricevladimir) [reply packages/ui-components/src/components/input/fileAvatarInput.tsx]: The function type definitions in the library forces this. `onDrop?: <T extends File>(acceptedFiles: T[], fileRejections: FileRejection[], event: DropEvent) => void;`

**Review by WissenIstNacht — APPROVED** [review]

**Pull request merged**